src/v/kafka/server/handlers/produce.cc

```diff
@@ -512,13 +512,20 @@ produce_topic(produce_ctx& octx, produce_request::topic& topic) {
                   .error_code = errc}));
         };
 
+        const bool is_transform_logs_topic
+          = config::shard_local_cfg().data_transforms_enabled()
```

Comment:
Let's remove the enablement check here, this is inline with audit and I think makes sense. (why can you write to this topic then enable transforms and not be able too?). The delete checks makes sense for cleanup if you disable wasm for some reason.

Reply:
done